app/controllers/application.js

```diff
@@ -8,44 +10,53 @@ export default Ember.Controller.extend({
     showUserOptions: false,
     search: Ember.computed.oneWay('controllers.search.q'),
 
-    stepFlash: function() {
+    stepFlash() {
         this.set('flashError', this.get('nextFlashError'));
         this.set('nextFlashError', null);
     },
 
-    aboutToTransition: function() {
+    aboutToTransition() {
         Ember.$(document).trigger('mousedown');
     },
 
-    resetDropdownOption: function(controller, option) {
-        controller.set(option, !controller.get(option));
-        if (controller.get(option)) {
-            Ember.$(document).on('mousedown.useroptions', function(e) {
+    // don't use this from other controllers..
+    resetDropdownOption(controller, option) {
+        this.set(option, !this.get(option));
+        if (this.get(option)) {
+            Ember.$(document).on('mousedown.useroptions', (e) => {
                 if (Ember.$(e.target).prop('tagName') === 'INPUT') {
                     return;
                 }
-                Ember.run.later(function() {
-                    controller.set(option, false);
+                Ember.run.later(() => {
+                    this.set(option, false);
                 }, 150);
                 Ember.$(document).off('mousedown.useroptions');
             });
         }
     },
 
-    currentPathChanged: function () {
+    _scrollToTop() {
         window.scrollTo(0, 0);
-    }.observes('currentPath'),
+    },
+
+    // TODO: remove observer & DOM mutation in controller..
+    currentPathChanged: observer('currentPath', function () {
+      Ember.run.scheduleOnce('afterRender', this, this._scrollToTop);
+    }),
 
     actions: {
-        search: function(query) {
+        search(q) {
             this.transitionToRoute('search', {
-              queryParams: {q: query, page: 1}
+              queryParams: {
+                q,
+                page: 1
+              }
             });
         },
 
-        toggleUserOptions: function() {
+        toggleUserOptions() {
             this.resetDropdownOption(this, 'showUserOptions');
-        },
-    },
```

Comment:
I prefer to leave trailing commas wherever possible, but is it invalid syntax now?